### PR TITLE
minimal implementation of REST API client

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		return nil, err
+	}
+
+	return client.doer.Do(req)
+}
+
+// Do performs a request and decodes the resulting json into the given
+// value. It's low-level, for testing/experimenting only; you should
+// usually use a higher level interface that builds on this.
+func (client *Client) Do(method, path string, body io.Reader, v interface{}) error {
```
Comment (niemeyer) [inline client/client.go]: Can we move this to export_test.go for the time being?  It ensures, at least for now, that we have more freedom changing the implementation without making many promises about how we're doing it. The rest of the client API (e.g. methods like SysInfo, etc) are our rock.

File: cmd/snap/main.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+func main() {
+	dirs.SetRootDir("/")
+
+	cli := client.New()
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 8, 1, ' ', 0)
+	fmt.Fprintln(w, "What\tError\tResult")
+	sysInfo, err := cli.SystemInfo()
```
Comment (niemeyer) [inline cmd/snap/main.go]: s/System/Sys/.. should fix tests.

**Comment by niemeyer** [comment]
Lovely small branches. Couple of trivials, but LGTM!

File: cmd/snap/main.go (hunk 2 above)
Comment (chipaca) [inline cmd/snap/main.go]: d'oh

File: client/client.go (hunk 1 above)
Comment (chipaca) [inline client/client.go]: Just in case, I'm understanding you to mean I should make Do private, and make it public in export_test. Let me know if I got this wrong.

Comment (niemeyer) [inline client/client.go]: Yeah, that's was it indeed.

**Comment by niemeyer** [comment]
Thanks! :+1:

File: client/client.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+// (exceptions are the icons/ endpoints obvs)
+type response struct {
+	Result     json.RawMessage `json:"result"`
+	Status     string          `json:"status"`
+	StatusCode int             `json:"status_code"`
+	Type       string          `json:"type"`
+}
+
+// SysInfo holds system information
+type SysInfo struct {
```
Comment (mvo5) [inline client/client.go]: Would it make sense to define these structs on the server (code) side or is that a stupid idea?

File: client/client.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	Flavor           string `json:"flavor"`
+	Release          string `json:"release"`
+	DefaultChannel   string `json:"default_channel"`
+	APICompatibility string `json:"api_compat"`
+	Store            string `json:"store,omitempty"`
+}
+
+// SysInfo gets system information from the REST API.
+func (client *Client) SysInfo() (*SysInfo, error) {
```
Comment (mvo5) [inline client/client.go]: Just a question out of curiosity, whats the plan to test this? The code looks like you replace client.doer with a mock one that just returns a mock json? Feel free to ignore this question is a subsequent pull request shows the testing :)

File: cmd/snap/main.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/ubuntu-core/snappy/client"
+	"github.com/ubuntu-core/snappy/dirs"
+)
+
+func main() {
+	dirs.SetRootDir("/")
```
Comment (mvo5) [inline cmd/snap/main.go]: Why is this needed?

File: client/client.go (hunk 3 above)
Comment (chipaca) [inline client/client.go]: It isn't a stupid idea; I'd just rather keep client self-contained (or at
least separate from server) at first.

**Comment by mvo5** [comment]
Thanks, looks great. Some silly questions inline but nothing that blocks this pull request, just curiosity on my side. So :+1:

File: cmd/snap/main.go (hunk 5 above)
Comment (chipaca) [inline cmd/snap/main.go]: Without it dirs.SnapdSocket is ""
